### 1045 - AddedOptionalProperty

> **Note:** Previous versions of this tool incorrectly flagged this as a breaking change - whilst care _does_ need to be taken when reviewing these - new **Optional** fields MUST be optional in request payloads for this to not be a breaking change.

**Description**: Checks whether a property is added to the model from the previous specification. The model includes all the models that referenced by any request or response.

**Cause**: A new optional field is added to the payload. This field is **Optional** therefore does not constitute a breaking change, in the event this field is **Required** in the new API version it should instead be marked as such. 

**Example**: Property `c` is being added into model `Parameters`.

Old specification
```json5
{
  "swagger": "2.0",
  "info": {
    "title": "swagger",
    "description": "The Azure Management API.",
    "version": "2016-12-01",
    ...
    ...
  "definitions": {
    ...
    ...
    "Parameters": {
      "properties": {
        "a": {
          "type": "string",
          "description": "Required. Property a."
        },
        "b": {
          "type": "string",
          "description": "Required. Enum Property b.",
          "enum": [ "b1", "b2" ]
        }
      },
      "description": "The parameters used when get operation."
    },
    ...  
```

New specification
```json5
{
  "swagger": "2.0",
  "info": {
    "title": "swagger",
    "description": "The Azure Management API.",
    "version": "2016-12-01",
    ...
    ...
  "definitions": {
    ...
    ...
    "Parameters": {
      "properties": {
        "a": {
          "type": "string",
          "description": "Required. Property a."
        },
        "b": {
          "type": "string",
          "description": "Required. Enum Property b.",
          "enum": [ "b1", "b2" ]
        },
        "c": {
          "type": "string",
          "description": "Property c."
        }
      },
      "description": "The parameters used when get operation."
    },
    ...  
```